R/quosure-join.R

```diff
@@ -0,0 +1,115 @@
+#' Join two `Quosure` objects
+#'
+#' Combine two `Quosure` object by merging their environments and the code.
+#' Not every `Quosure` objects can be combined:
+#' - if their environments contains objects of the same name but not identical
+#' - if `y` has unique code element placed before common element. This means that `y`
+#' in the environment of the `y` was evaluated some extra code before which can influence
+#' reproducibility
+#' - more cases to be done
```

Comment:
make a reference to `check_joinable` and maintain good docs in one place only

Reply:
check_joinable is a private utility. Is it ok to expose this function in the public docs?

Reply:
You can still have a roxygen documentation for private functions and manage public/private in the pkgdown config file.

Reply:
Yes we can, but the last time we discussed this we agreed that:
1. all private functions should have @keyword internal
2. private functions shouldn't be exposed in the public docs (to avoid users using private functions)
3. internal functions not listed in pkgdown

We can make a reference in the docs, but that's against (2). I'm totally against listing .check_joinable in the pkgdown

Reply:
new issue https://github.com/insightsengineering/teal.code/issues/71